File: JavaSource/src/main/java/ru/akhcheck/patterns/observer/CountingObserver.java
package ru.akhcheck.patterns.observer;

import java.util.concurrent.atomic.AtomicInteger;

/**
 * Ничего не печатает, только считает.
 *
 * <p>Показывает, что подписчики независимы: источник один, а делают они разное
 * и друг про друга не знают. Счётчик атомарный, потому что сообщения приходят
 * из нескольких потоков.
 */
public class CountingObserver implements Observer {
    private final AtomicInteger count = new AtomicInteger();

    @Override
    public void update(String message) {
        count.incrementAndGet();
    }

    public int getCount() {
        return count.get();
    }
}

File: Observer/python-source/observers/CountingObserver.py
from __future__ import annotations

from .Observer import Observer


class CountingObserver(Observer):
    """Ничего не печатает, только считает.

    Показывает, что подписчики независимы: источник один, а делают они разное
    и друг про друга не знают.
    """

    def __init__(self) -> None:
        self.count = 0

    def update(self, message: str) -> None:
        self.count += 1
